evm/bridge-token-factory/contracts/BridgeTokenFactory.sol

```diff
@@ -5,104 +5,43 @@ import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/acce
 import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
 import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 import "./BridgeToken.sol";
 import "./SelectivePausableUpgradable.sol";
 import "./Borsh.sol";
+import "./BridgeTypes.sol";
 
 contract BridgeTokenFactory is
     UUPSUpgradeable,
     AccessControlUpgradeable,
     SelectivePausableUpgradable
 {
-    enum WhitelistMode {
-        NotInitialized,
-        Blocked,
-        CheckToken,
-        CheckAccountAndToken
-    }
-
     // We removed ProofConsumer from the list of parent contracts and added this gap
     // to preserve storage layout when upgrading to the new contract version.
     uint256[54] private __gap;
+    using SafeERC20 for IERC20;
 
-    mapping(address => string) private _ethToNearToken;
-    mapping(string => address) private _nearToEthToken;
-    mapping(address => bool) private _isBridgeToken;
-
-    mapping(string => WhitelistMode) private _whitelistedTokens;
-    mapping(bytes => bool) private _whitelistedAccounts;
-    bool private _isWhitelistModeEnabled;
+    mapping(address => string) public ethToNearToken;
+    mapping(string => address) public nearToEthToken;
+    mapping(address => bool) public isBridgeToken;
 
     address public tokenImplementationAddress;
     address public nearBridgeDerivedAddress;
     uint8 public omniBridgeChainId;
 
+    uint256[3] private __gapForRemovedFields;
```

Comment:
This implies we intend to upgrade some of the contracts? I thought we will deploy a new one

Reply:
Yeah, I think it is better to deploy a new one, but we should be careful when changing the owner of the already deployed tokens. Also, we need to make sure that the old factory has this functionality.

Reply:
Why do we need a gap then? I think we'll deploy a new contract and then phase out the old ones?